prism.el

```diff
@@ -907,44 +905,53 @@ arguments to set the same faces."
       (customize-save-variable var (symbol-value var))))
   (customize-save-variable 'prism-colors prism-colors))
 
-(cl-defun prism-modify-colors (&key num colors desaturations lightens &allow-other-keys)
-  "Return list of NUM colors modified according to DESATURATIONS and LIGHTENS."
-  (cl-flet ((modify-color (color desaturate lighten)
+(cl-defun prism-modify-colors (&key num colors desaturations lightens opacities &allow-other-keys)
+  "Return list of NUM COLORS modified according to DESATURATIONS, LIGHTENS, and OPACITIES."
+  (cl-flet ((modify-color (color desaturate lighten opacify)
                           (--> color
-                            (if (> desaturate 0)
-                                (color-desaturate-name it desaturate)
-                              it)
-                            (if (> lighten 0)
-                                (color-lighten-name it lighten)
-                              it)
-                            ;; FIXME: It seems that these two functions called in sequence
-                            ;; always modify the color, e.g. #ff2afc becomes #fe29fb.
-                            (color-name-to-rgb it)
-                            (-let (((r g b) it))
-                              (color-rgb-to-hex r g b 2)))))
+                               (if (> desaturate 0)
+                                   (color-desaturate-name it desaturate)
+                                 it)
+                               (if (> lighten 0)
+                                   (color-lighten-name it lighten)
+                                 it)
+                               (if (> opacify 0)
+                                   (prism-opacify-name it opacify)
+                                 it)
     (when (< (length desaturations) num)
       (setf desaturations (prism-expand-list num desaturations)))
     (when (< (length lightens) num)
       (setf lightens (prism-expand-list num lightens)))
+    (when (< (length opacities) num)
+      (setf opacities (prism-expand-list num opacities)))
     (cl-loop for i from 0 below num
              for desaturate = (nth i desaturations)
              for lighten = (nth i lightens)
-             collect (modify-color (nth i colors) desaturate lighten))))
+             for opacify = (nth i opacities)
+             collect (modify-color (nth i colors) desaturate lighten opacify))))
 
 (defun prism-blend (a b alpha)
   "Return color A blended with color B by amount ALPHA."
-  (cl-flet ((blend (a b alpha)
-                   (+ (* alpha a) (* b (- 1 alpha)))))
-    (-let* (((ar ag ab) (color-name-to-rgb a))
-            ((br bg bb) (color-name-to-rgb b)))
-      (color-rgb-to-hex (blend ar br alpha)
-                        (blend ag bg alpha)
-                        (blend ab bb alpha)))))
+  (if (some #'(lambda (color) (member color '(unspecified "unspecified-fg" "unspecified-bg")))
+            `(,a ,b))
+      nil
```

Comment:
If this is a bug that needs to be fixed, please do it separate from this PR.

Reply:
Can do, I just bundled it here because it was affecting my testing.